front-end/adminPanel.py

```diff
@@ -0,0 +1,160 @@
+        self.user_tab.grid_rowconfigure(0, weight=1)
+        self.user_tab.grid_columnconfigure(0, weight=1)
+        tk.Label(self.main_tab, text="Panel Admina - Główna", font=("Arial", 24)).pack(pady=20)
+        tk.Label(self.main_tab, text="Witaj w panelu administracyjnym!", font=("Arial", 14)).pack(pady=10)
+
+    def create_user_tab(self):
+        self.user_tab.grid_rowconfigure(0, weight=1)
+        self.user_tab.grid_columnconfigure(0, weight=1)
+        self.create_user_tab_view(self.user_tab)
+
+    def create_user_tab_view(self, frame):
+        frame.grid_rowconfigure(0, weight=1)
+        frame.grid_rowconfigure(1, weight=10)
+        frame.grid_columnconfigure(0, weight=1)
+        frame.grid_columnconfigure(1, weight=4)
+
+        tk.Label(frame, text="Zarządzanie użytkownikami", font=("Arial", 24)).grid(row=0, column=0, columnspan=2, pady=20)
+
+        buttons_frame = tk.Frame(frame)
+        buttons_frame.grid(row=1, column=0, sticky="nsew", padx=10, pady=10)
+
+        actions = [
+            ("Dodaj użytkownika", self.add_user),
+            ("Usuń użytkownika", self.delete_user),
+        ]
+
+        buttons_frame.grid_rowconfigure(0, weight=1)
+        buttons_frame.grid_rowconfigure(len(actions) + 1, weight=1)
+        buttons_frame.grid_columnconfigure(0, weight=1)
+        buttons_frame.grid_columnconfigure(2, weight=1)
+
+        for idx, (label, command) in enumerate(actions):
+            tk.Button(buttons_frame, text=label, font=("Arial", 16), bg="gray", fg="white",
+                      width=15, height=2, command=command).grid(row=idx + 1, column=1, pady=10)
+
+        style = ttk.Style()
+        style.configure("Treeview", font=("Arial", 14), rowheight=40)
+        style.configure("Treeview.Heading", font=("Arial", 16, "bold"))
+
+        users_frame = tk.Frame(frame)
+        users_frame.grid(row=1, column=1, sticky="nsew", padx=10, pady=10)
+
+        tk.Label(users_frame, text="Lista użytkowników", font=("Arial", 18)).pack(pady=10)
+
+        self.users_listbox = ttk.Treeview(users_frame, columns=("Username",), show="headings")
+        self.users_listbox.heading("Username", text="Nazwa użytkownika")
+        self.users_listbox.pack(fill="both", expand=True)
+
+        self.load_users()
+
+    def load_users(self):
+        users = self.request_handler.get_all_users()
+        if users:
+            self.users_listbox.delete(*self.users_listbox.get_children())
+            for user in users:
+                self.users_listbox.insert("", "end", values=(user['username'],))
+
+    def add_user(self):
+        form = tk.Toplevel(self)
+        form.title("Dodaj użytkownika")
+
+        tk.Label(form, text="Nazwa użytkownika:", font=("Arial", 14)).grid(row=0, column=0, padx=10, pady=10)
+        username_entry = tk.Entry(form, font=("Arial", 14))
+        username_entry.grid(row=0, column=1, padx=10, pady=10)
+
+        tk.Label(form, text="Hasło:", font=("Arial", 14)).grid(row=1, column=0, padx=10, pady=10)
```

Comment:
Nie hasło a PIN, plus gdzie jest rola użytkownika? Dodatkowo zwykły użytkownik powinien mieć pin długości dokładnie 4 znaków jak dobrze pamiętam a tutaj nie ma żadnej walidacji przez co można jako hasło wpisać nawet stringa mimo, że powinniśmy przyjmować tylko liczby naturalne...

Reply:
Spoko, chciałem przygotować już podstawowe funkcje dla obsługi użytkownika byśmy mogli podłączyć backend.
ale ta zajme się "Dodatkowo zwykły użytkownik powinien mieć pin długości dokładnie 4 znaków jak dobrze pamiętam a tutaj nie ma żadnej walidacji przez co można jako hasło wpisać nawet stringa mimo, że powinniśmy przyjmować tylko liczby naturalne..."

Reply:
Kurde zrobiłem zeby byly te 4 cyfry ale musimy ustalic na ile my robimy walidacje a na ile robi ja backen - bo walidacja chyba powinna byc przeprowadzana przez backend?
u nas to bardziej by np by ilosc znakow niewychodziła po za ramke